web_programming/web_scrapping.py

```diff
@@ -0,0 +1,144 @@
+
+    meta_info = {}
+    language = str(soup.html.get("lang"))
+    if language == "None":
+        meta_info["language"] = "Not mentioned"
+    else:
+        meta_info["language"] = language
+    charset = str(soup.meta.get("charset"))
+    if charset == "None":
+        meta_info["charset"] = "Not mentioned"
+    else:
+        meta_info["charset"] = charset
+
+    try:
+        title = soup.find("meta", property="og:title")["content"]
+        if title:
+            meta_info["title"] = title
+    except KeyError:
+        try:
+            title = soup.find("meta", attrs={"name": "title"})["content"]
+            if title:
+                meta_info["title"] = title
+        except KeyError:
+            meta_info["title"] = "Not mentioned"
+
+    try:
+        type_ = soup.find("meta", property="og:type")["content"]
+        if type_:
+            meta_info["type"] = type_
+    except KeyError:
+        try:
+            type_ = soup.find("meta", attrs={"name": "type"})["content"]
+            if type_:
+                meta_info["type"] = type_
+        except KeyError:
+            meta_info["type"] = "Not mentioned"
+
+    try:
+        description = soup.find("meta", property="og:description")["content"]
+        if description:
+            meta_info["description"] = description
+    except KeyError:
+        try:
+            description = soup.find("meta", attrs={"name": "description"})["content"]
+            if description:
+                meta_info["description"] = description
+        except KeyError:
+            meta_info["description"] = "Not mentioned"
+
+    try:
+        site_name = soup.find("meta", property="og:site_name")["content"]
+        if site_name:
+            meta_info["site_name"] = site_name
+    except KeyError:
+        try:
+            site_name = soup.find("meta", attrs={"name": "site_name"})["content"]
+            if site_name:
+                meta_info["site_name"] = site_name
+        except KeyError:
+            meta_info["site_name"] = "Not mentioned"
+
+    try:
+        image = soup.find("meta", property="og:image")["content"]
+        if image:
+            meta_info["image"] = image
+    except KeyError:
+        try:
+            image = soup.find("meta", attrs={"name": "image"})["content"]
+            if image:
+                meta_info["image"] = image
+        except KeyError:
+            meta_info["image"] = "Not mentioned"
+
+    try:
+        keywords = soup.find("meta", property="og:keywords")["content"]
+        if keywords:
+            meta_info["keywords"] = keywords
+    except KeyError:
+        try:
+            keywords = soup.find("meta", attrs={"name": "keywords"})["content"]
+            if keywords:
+                meta_info["keywords"] = keywords
+        except KeyError:
+            meta_info["keywords"] = "Not mentioned"
+
+    if meta_info["title"] == "Not mentioned":
+        title = soup.find("title").text
+        meta_info["title"] = title
```

Comment:
Don't create variables that only live for two lines unless the lines are really long (~88 characters).
```suggestion
        meta_info["title"] = soup.find("title").text
```